### PR TITLE
4 create some visualization

### PR DESCRIPTION
**Pull request by totuk2**
packer.py presents the implementation of the packaging algorythm with visualisation and textualisation options.

### CODE REVIEW

File: packer.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+def refresh_bin_types(bin_types):
+    for bin in bin_types:
+        packer.add_bin(bin)
+def execute_packing(items_to_fit: list, visualize=True, export_img=False, textualize=True) -> list:
+    fitted_items = []   # list of solutions
+    tree = Tree("Packing list:", highlight=True, hide_root=True)
+    while items_to_fit:
+        packing_efficacy = 0
+
+        global packer
```
Comment (jedrus2000) [inline packer.py:65]: Avoid global ! Better pass it as parameter, return it from function.

File: py3dbp/main.py [hunk 2]
```diff
@@ -1,5 +1,12 @@
 from .constants import RotationType, Axis
 from .auxiliary_methods import intersect, set_to_decimal
+from copy import deepcopy
+
+
+# required to plot a representation of Bin and contained items
+import matplotlib.pyplot as plt
```
Comment (jedrus2000) [inline py3dbp/main.py:7]: Remove plot function and dependencies from this main package. Create this functions outside, keep packer classes clean.

**Review by jedrus2000 — CHANGES_REQUESTED** [review]

File: packer.py (hunk 1 above)
Comment (totuk2) [reply packer.py]: Ok. Passed as parameter.

File: py3dbp/main.py (hunk 2 above)
Comment (totuk2) [reply py3dbp/main.py]: Moved to auxiliary_methods.py

File: py3dbp/auxiliary_methods.py [hunk 3]
```diff
@@ -33,3 +35,47 @@ def set_to_decimal(value, number_of_decimals):
...
+    yy = [y, y+dy, y+dy, y, y]
+    kwargs = {'alpha': 1, 'color': color, 'lw': lw}
+    ax.plot3D(xx, yy, [z]*5, **kwargs)
+    ax.plot3D(xx, yy, [z+dz]*5, **kwargs)
+    ax.plot3D([x, x], [y, y], [z, z+dz], **kwargs)
+    ax.plot3D([x, x], [y+dy, y+dy], [z, z+dz], **kwargs)
+    ax.plot3D([x+dx, x+dx], [y+dy, y+dy], [z, z+dz], **kwargs)
+    ax.plot3D([x+dx, x+dx], [y, y], [z, z+dz], **kwargs)
+
+def plotBoxAndItems(bin, export_img, title=""):
```
Comment (jedrus2000) [inline py3dbp/auxiliary_methods.py:51]: Changes looks good ! However, name functions with with Python convention: `plot_box_and_items`.
I think pyCharm is giving hints for it. Look at them !

image

Comment (totuk2) [reply py3dbp/auxiliary_methods.py]: OK

**Review by jedrus2000 — APPROVED** [review]

**Pull request merged**